stdlib/in_channel.mli

```diff
@@ -0,0 +1,164 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *)
+(*                                                                        *)
+(*   Copyright 2019 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+
+(** Input channels.
+
+    @since 4.10 *)
+
+type t = in_channel
+
+val stdin : t
+(** Standard output. *)
+
+(** {2 General input functions} *)
+
+type 'a open_function =
+  ?create:bool -> (* default: false *)
+  ?binary:bool -> (* default: false *)
+  'a
```

Comment:
To give a concrete example [here is](https://b0-system.github.io/odig/doc@odig.dark/b0/B0_std/Os/File/index.html#val-with_tmp_fd) a function that uses its own defaults for (`Unix`) open flags but still allows the client to override them. It would be quite very annoying to expose and implement this function with the `type 'a open_function` way of doing this PR suggests.

Reply:
Makes sense, I will switch back to having an `open_flag` type.

Reply:
I added a flag type back, with different flags for input and output channels. I left in the "non-blocking" flag even if it does not fit very well with an "option"-based API... I'm not sure what to do about that.